internal/filter/types/watchlist/watchlist.go

```diff
@@ -100,6 +101,7 @@ func (c *Filter) handleEvent(event watch.Event) {
 	}
 
 	if cachedObject || wasSnapshot || wasDeleted {
+		metrics.RecordDataEntry(ctx, entry)
 		c.out <- entry
 	}
```

Comment:
We might want to add a metric below this captures that the entry was an out of data object or duplicate.